depthai_nodes/ml/parsers/parser.py

```diff
@@ -1,90 +1,61 @@
-from typing import Dict, List, Union
+from typing import Dict
 
 import depthai as dai
 
+from . import *
+from .base_parser import BaseParser
+from .utils import decode_head
+
 
 class Parser(dai.node.ThreadedHostNode):
-    """Base class for all parsers."""
+    """General interface for instantiating parsers based on the provided model
+    archive."""
```

Comment:
`Parser` class is now used to initialize parser(s) from a NN Archive. I'd rename it to better indicate its function, e.g. to `ParserInitializer` or `NNArchiveParser` etc.

Reply:
I would leave it as is beacause i think its way more readable to instantiate it like this (in a case where NeuralNetworkWithParser isnt used). I would say you remember to instantiate a *Parser* before *ParserInitializer*.

```
pipeline.create(Parser).build(nn_archive)
```

Reply:
@klemen1999 @kkeroo for opinion

Reply:
Maybe also `ParsersGenerator`?

Reply:
Hm, I would also go with different names (some of the Jakob's proposals)